.github/workflows/main.yml

```diff
@@ -5,6 +5,15 @@ on:
     branches: [ "master" ]
   pull_request:
     branches: [ "master" ]
+  workflow_dispatch:
+  schedule:
+    # every 1st day of month at 00:00 UTC
+    - cron: "0 0 1 * *"
+
+permissions:
```

Comment:
Can you explain this section to me? Why does this workflow need non-default permissions? What does `checks: write` and `pull-requests: write` mean?